src/tests/ftest/container/api_basic_attribute.py

```diff
@@ -0,0 +1,294 @@
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a ContainerAttributeTest object."""
+        super().__init__(*args, **kwargs)
+        self.attr_name = None
+        self.attr_val = None
+        self.attr_dict = {}
+        self.expected_result = None
+
+    def verify_list_attr(self, in_dict, out_size, out_buf, is_async=True):
+        """Verify list_attr output; size and buf.
+
+        Args:
+            in_dict (dict): Dictionary that stores the Name-Value pairs that's used
+                in set_attr().
+            out_size (int): Size returned by list_attr().
+            out_buf (str): Buffer returned by list_attr(). It stores the names in the
+                following format.
+                b'Name1\x00Name2\x00Name3\x00\x00'
+                Each name has following \x00 and one \x00 at the end.
+            is_async (bool): Whether list_attr() was called with asynchronous mode.
+                Defaults to True.
+        """
+        self.log.info("in_dict = %s", in_dict)
+        self.log.info("out_size = %d", out_size)
+        self.log.info("out_buf = %s", out_buf)
+
+        # Create the list of names from the input dictionary and the list_attr() output
+        # and sort them.
+        in_names = sorted(list(in_dict.keys()))
+        out_names = sorted(out_buf.split(b"\x00")[:-2])
+
+        self.log.info("in_names = %s", in_names)
+        self.log.info("out_names = %s", out_names)
+
+        # In order to verify the size, calculate the total number of characters of all
+        # the names in the input dictionary.
+        total_str_len = sum(len(name) for name in in_names)
+        # Add the number of \x00 because the size includes them.
+        total_zero_buf_len = len(in_names)
+        # Size from the async mode is one larger than sync. It correctly counts the
+        # number of "\x00"s.
+        if is_async:
+            total_zero_buf_len += 1
+        expected_size = total_str_len + total_zero_buf_len
+
+        self.log.info("total_str_len = %d", total_str_len)
+        self.log.info("total_zero_buf_len = %d", total_zero_buf_len)
+        self.log.info("expected_size = %d", expected_size)
+
+        msg = "Unexpected size from list_attr! Expected = {}; Actual = {}".format(
+            expected_size, out_size)
+        self.assertEqual(expected_size, out_size, msg)
+
+        msg = ("Unexpected output from list_attr! Expected = {}; Actual raw = {};"
+               "Actual in list format = {}".format(in_names, out_buf, out_names))
+        self.assertEqual(out_names, in_names, msg)
+
+    def verify_get_attr(self, indata, outdata):
+        """Verify the Attributes value after get_attr.
+
+        Args:
+            indata (dict): Name-value pairs that were used for set-attr.
+            outdata (dict): Name-value pairs that were obtained from get-attr.
+        """
+        decoded = {}
+        for key, val in outdata.items():
+            decoded[key.decode()] = val
```

Comment:
Just FYI, this also works
```suggestion
        decoded = {key.decode(): val for key, val in outdata.items()}
```

Reply:
Okay, thanks!